api/src/auth/auth_utils.py

```diff
@@ -19,6 +19,7 @@ def get_app_security_scheme() -> dict[str, Any]:
         "ApiKeyAuth": {"type": "apiKey", "in": "header", "name": "X-Auth"},
         "ApiJwtAuth": {"type": "apiKey", "in": "header", "name": "X-SGG-Token"},
         "InternalApiJwtAuth": {"type": "apiKey", "in": "header", "name": "X-SGG-Internal-Token"},
+        "ApiGatewayKeyAuth": {"type": "apiKey", "in": "header", "name": "X-API-Key"},
```

Comment:
Instead of `ApiGatewayKeyAuth` can we call this something that doesn't reference the AWS service? Something like `ApiUserKeyAuth` maybe? The user/our own API doesn't care that it's API gateway.